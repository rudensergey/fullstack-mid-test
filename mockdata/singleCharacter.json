{
  "data": {
    "id": 48,
    "name": "Black Rick",
    "status": "Alive",
    "species": "Human",
    "gender": "Male",
    "image": "https://rickandmortyapi.com/api/character/avatar/48.jpeg",
    "origin": {
      "id": null,
      "name": "unknown",
      "type": null,
      "dimension": null,
      "residents": [
        {
          "status": "Alive",
          "species": "Human",
          "type": "",
          "name": "Rick Sanchez",
          "id": 1,
          "gender": "Male",
          "image": "https://rickandmortyapi.com/api/character/avatar/48.jpeg"
        },
        {
          "status": "Alive",
          "species": "Human",
          "type": "",
          "name": "Morty Smith",
          "id": 2,
          "gender": "Male",
          "image": "https://rickandmortyapi.com/api/character/avatar/48.jpeg"
        }
      ]
    },
    "location": {
      "type": "Space station",
      "id": 3,
      "name": "Citadel of Ricks",
      "dimension": "unknown",
      "residents": [
        {
          "status": "Alive",
          "species": "Human",
          "type": "",
          "name": "Rick Sanchez",
          "id": 1,
          "gender": "Male",
          "image": "https://rickandmortyapi.com/api/character/avatar/48.jpeg"
        },
        {
          "status": "Alive",
          "species": "Human",
          "type": "",
          "name": "Morty Smith",
          "id": 2,
          "gender": "Male",
          "image": "https://rickandmortyapi.com/api/character/avatar/48.jpeg"
        }
      ]
    },
    "episode": [
      {
        "id": 22,
        "name": "The Rickshank Rickdemption",
        "air_date": "April 1, 2017",
        "episode": "S03E01",
        "characters": [
          {
            "id": 1,
            "name": "Rick Sanchez",
            "status": "Alive",
            "species": "Human",
            "gender": "Male",
            "image": "https://rickandmortyapi.com/api/character/avatar/48.jpeg"
          },
          {
            "id": 2,
            "name": "Morty Smith",
            "status": "Alive",
            "species": "Human",
            "gender": "Male",
            "image": "https://rickandmortyapi.com/api/character/avatar/48.jpeg"
          }
        ]
      },
      {
        "id": 28,
        "name": "The Ricklantis Mixup",
        "air_date": "September 10, 2017",
        "episode": "S03E07",
        "characters": [
          {
            "id": 1,
            "name": "Rick Sanchez",
            "status": "Alive",
            "species": "Human",
            "gender": "Male",
            "image": "https://rickandmortyapi.com/api/character/avatar/48.jpeg"
          }
        ]
      }
    ]
  }
}
