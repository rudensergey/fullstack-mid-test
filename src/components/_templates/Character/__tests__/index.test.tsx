// absolute
import { render } from "@testing-library/react";

// components
import CharacterTemplate from "@templates/Character";

// mock
import { data } from "@mock/singleCharacter.json";

describe("Character template", () => {
  it("snapshot", () => {
    const { container } = render(<CharacterTemplate title="test" character={data} />);
    expect(container).toMatchSnapshot();
  });
});
