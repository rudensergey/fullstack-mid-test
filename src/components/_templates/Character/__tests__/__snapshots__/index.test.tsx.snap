// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character template snapshot 1`] = `
<div>
  <div
    class="Backgroundstyled__ImageWrapper-sc-7ewg2c-0 fBhSrK"
  >
    <div
      style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
    >
      <img
        alt="wall with Rick"
        data-nimg="fill"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
      />
      <noscript />
    </div>
    <div
      class="Characterstyled__Wrapper-sc-18gu1ab-0 bvyWpx"
    >
      <p
        class="Characterstyled__Title-sc-18gu1ab-1 gLtUAz"
      >
        Rick and Morty
      </p>
      <button
        class="Buttonstyled__Button-sc-1q754k2-1 Buttonstyled__Button-sc-1q754k2-0 bRFmVW bDelgz"
      >
        &lt; Back to charact listing
      </button>
      <div
        class="Characterstyled__CharacterCore-sc-18gu1ab-2 jnRYxB"
      >
        <div
          class="Characterstyled__ImageWrapper-sc-18gu1ab-3 kXGERR"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAwIiBoZWlnaHQ9IjIwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="Black Rick"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
            <noscript />
          </div>
        </div>
        <div
          class="Characterstyled__CharacterCoreWrapper-sc-18gu1ab-4 bgsObY"
        >
          <p
            class="Characterstyled__Name-sc-18gu1ab-5 dRmzYL"
          >
            Black Rick
          </p>
          <p
            class="Descriptionstyled__Text-sc-1d7kuch-0 gFVcQU"
          >
            <span
              class="Descriptionstyled__Span-sc-1d7kuch-1 erJcwk"
            >
              Status
            </span>
            : Alive
          </p>
          <p
            class="Descriptionstyled__Text-sc-1d7kuch-0 gFVcQU"
          >
            <span
              class="Descriptionstyled__Span-sc-1d7kuch-1 erJcwk"
            >
              Origin
            </span>
            : unknown
          </p>
        </div>
      </div>
      <div
        class="Infostyled__Wrapper-sc-1c5crsb-0 idGidK"
      >
        <p
          class="Infostyled__Title-sc-1c5crsb-1 bTsvHP"
        >
          Location Details
        </p>
        <p
          class="Descriptionstyled__Text-sc-1d7kuch-0 gFVcQU"
        >
          <span
            class="Descriptionstyled__Span-sc-1d7kuch-1 erJcwk"
          >
            Name
          </span>
          : Citadel of Ricks
        </p>
        <p
          class="Descriptionstyled__Text-sc-1d7kuch-0 gFVcQU"
        >
          <span
            class="Descriptionstyled__Span-sc-1d7kuch-1 erJcwk"
          >
            Type
          </span>
          : Space station
        </p>
        <p
          class="Descriptionstyled__Text-sc-1d7kuch-0 gFVcQU"
        >
          <span
            class="Descriptionstyled__Span-sc-1d7kuch-1 erJcwk"
          >
            Dimension
          </span>
          : unknown
        </p>
        <p
          class="Descriptionstyled__Text-sc-1d7kuch-0 gFVcQU"
        >
          <span
            class="Descriptionstyled__Span-sc-1d7kuch-1 erJcwk"
          >
            No. of Residents
          </span>
          : 2
        </p>
      </div>
      <div
        class="Infostyled__Wrapper-sc-1c5crsb-0 idGidK"
      >
        <p
          class="Infostyled__Title-sc-1c5crsb-1 bTsvHP"
        >
          Episodes: 2
        </p>
        <p
          class="Descriptionstyled__Text-sc-1d7kuch-0 gFVcQU"
        >
          <span
            class="Descriptionstyled__Span-sc-1d7kuch-1 erJcwk"
          >
            First appearance
          </span>
          : The Rickshank Rickdemption:S03E01
        </p>
        <p
          class="Descriptionstyled__Text-sc-1d7kuch-0 gFVcQU"
        >
          <span
            class="Descriptionstyled__Span-sc-1d7kuch-1 erJcwk"
          >
            First appearance air date
          </span>
          : April 1, 2017
        </p>
        <p
          class="Descriptionstyled__Text-sc-1d7kuch-0 gFVcQU"
        >
          <span
            class="Descriptionstyled__Span-sc-1d7kuch-1 erJcwk"
          >
            First appearance character count
          </span>
          : 2
        </p>
        <p
          class="Descriptionstyled__Text-sc-1d7kuch-0 gFVcQU"
        >
          <span
            class="Descriptionstyled__Span-sc-1d7kuch-1 erJcwk"
          >
            Last appearance
          </span>
          : The Ricklantis Mixup:S03E07
        </p>
        <p
          class="Descriptionstyled__Text-sc-1d7kuch-0 gFVcQU"
        >
          <span
            class="Descriptionstyled__Span-sc-1d7kuch-1 erJcwk"
          >
            Last appearance air date
          </span>
          : September 10, 2017
        </p>
        <p
          class="Descriptionstyled__Text-sc-1d7kuch-0 gFVcQU"
        >
          <span
            class="Descriptionstyled__Span-sc-1d7kuch-1 erJcwk"
          >
            Last appearance character count
          </span>
          : 1
        </p>
      </div>
    </div>
  </div>
</div>
`;
